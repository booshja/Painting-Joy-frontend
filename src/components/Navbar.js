// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
// components
import NavLink from "./NavLink";
// context
import CartContext from "../context/CartContext";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledNav = styled.nav`
    display: none;
    align-items: center;
    justify-content: space-between;
    padding-bottom: 1rem;
    width: 55%;

    ${breakpoints("display", "", [{ 1024: "flex" }])}
`;

const Navbar = () => {
    // set up context
    const { cart } = useContext(CartContext);

    return (
        <StyledNav>
            <NavLink to="/" name="Home" />
            <NavLink to="/murals" name="Murals" />
            {/* <NavLink to="/more-art" name="More Art" /> */}
            <NavLink to="/contact" name="Contact" />
            <NavLink to="/store" name="Store" />
            {/* <NavLink to="/cart" name={`Cart (${cart.length})`} /> */}
        </StyledNav>
    );
};

export default Navbar;
