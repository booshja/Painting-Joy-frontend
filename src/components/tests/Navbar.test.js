import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Navbar from "../Navbar";
import CartContext from "../../context/CartContext";

/** Smoke Test */
it("renders without crashing, cart > 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3, 4] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );
});

it("renders without crashing, cart = 0", () => {
    render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot, cart > 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3, 4] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot, cart = 0", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly, cart > 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [1, 2, 3, 4] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact");
    getByText("Store");
    getByText("Cart (4)");
});

it("renders correctly, cart = 0", () => {
    const { getByText } = render(
        <BrowserRouter>
            <CartContext.Provider value={{ cart: [] }}>
                <Navbar />
            </CartContext.Provider>
        </BrowserRouter>
    );

    getByText("Home");
    getByText("Murals");
    getByText("Contact");
    getByText("Store");
    getByText("Cart (0)");
});
