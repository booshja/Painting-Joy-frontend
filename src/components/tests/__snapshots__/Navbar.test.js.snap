// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot, cart = 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-gtsrHT lprbgz"
  >
    <a
      aria-current="page"
      class="sc-bdnxRM jTcfkt active"
      href="/"
    >
      Home
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/murals"
    >
      Murals
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/contact"
    >
      Contact
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/store"
    >
      Store
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/cart"
    >
      Cart (0)
    </a>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot, cart > 0 1`] = `
<DocumentFragment>
  <nav
    class="sc-gtsrHT lprbgz"
  >
    <a
      aria-current="page"
      class="sc-bdnxRM jTcfkt active"
      href="/"
    >
      Home
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/murals"
    >
      Murals
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/contact"
    >
      Contact
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/store"
    >
      Store
    </a>
    <a
      class="sc-bdnxRM jTcfkt"
      href="/cart"
    >
      Cart (4)
    </a>
  </nav>
</DocumentFragment>
`;
